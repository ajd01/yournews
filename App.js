import React, { useReducer } from 'react'
import { View, Text, StyleSheet, Dimensions } from 'react-native'
import Overview from './components/Overview'
import Detail from './components/Details'
import Back from './components/Back'
import {
  Context,
  InitialState,
  Reducer
} from './store'

export default function App () {
  const [store, dispatch] = useReducer(Reducer, InitialState)

  const { selected } = store

  return (
    <Context.Provider value={{ store, dispatch }}>
      <View
        style={{
          flex: 1,
          backgroundColor: '#c4e248'
        }}
      >
        <Text style={styles.Header}>Your News</Text>
        {selected
          ? <Back />
          : <View />}
        <View style={styles.Content}>
          {selected
            ? <Detail />
            : <Overview />}
        </View>
      </View>
    </Context.Provider>

  )
}

const styles = StyleSheet.create({
  Header: {
    paddingTop: 5,
    paddingBottom: 5,
    fontSize: 45,
    fontWeight: 'bold',
    backgroundColor: '#000',
    color: '#fff',
    textAlign: 'center',
    width: Dimensions.get('window').width,
    position: 'relative',
    zIndex: 98
  },
  Content: {
    marginTop: 60
  }
})
