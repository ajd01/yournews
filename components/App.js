import React, { useContext } from 'react'
import { View, Text, StyleSheet, Dimensions } from 'react-native'
import Overview from './Overview'
import Detail from './Details'
import Back from './Back'
import { Context } from '../store'

export default function App () {
  const { store } = useContext(Context)

  const { selected } = store

  return (
    <View
      style={{
        flex: 1,
        backgroundColor: '#c4e248'
      }}
    >
      <Text style={styles.Header}>Your News</Text>
      {selected
        ? <Back />
        : <View />}
      <View style={styles.Content}>
        {selected
          ? <Detail />
          : <Overview />}
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  Header: {
    paddingTop: 5,
    paddingBottom: 5,
    fontSize: 3,
    fontWeight: 'bold',
    backgroundColor: '#000',
    color: '#fff',
    textAlign: 'center',
    width: Dimensions.get('window').width,
    position: 'relative',
    zIndex: 98
  },
  Content: {
    marginTop: 60
  }
})
