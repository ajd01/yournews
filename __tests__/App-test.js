import React from 'react'
import App from '../App'
import renderer from 'react-test-renderer'

jest.mock('../components/Overview', () => {
  return {
    __esModule: true,
    Overview: true,
    default: () => {
      return <div />
    }
  }
})
jest.mock('../components/Details', () => {
  return (
    <div />
  )
})
jest.mock('../components/Back', () => {
  return (
    <div />
  )
})

describe('App Component', () => {
  it('renders screen', () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
