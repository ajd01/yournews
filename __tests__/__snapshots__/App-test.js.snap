// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#c4e248",
      "flex": 1,
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "#000",
        "color": "#fff",
        "fontSize": 45,
        "fontWeight": "bold",
        "paddingBottom": 5,
        "paddingTop": 5,
        "position": "relative",
        "textAlign": "center",
        "width": 750,
        "zIndex": 98,
      }
    }
  >
    Your News
  </Text>
  <View />
  <View
    style={
      Object {
        "marginTop": 60,
      }
    }
  >
    <div />
  </View>
</View>
`;
